ui/app/services/wizard.js

```diff
@@ -27,30 +27,45 @@ export default Service.extend(DEFAULTS, {
     if (this.storageHasKey(STORAGE_KEYS.COMPONENT_STATE)) {
       this.set('componentState', this.getExtState(STORAGE_KEYS.COMPONENT_STATE));
     }
-    let stateNodes = TutorialMachine.getStateNodes(this.get('currentState'));
+    let stateNodes = TutorialMachine.getStateNodes(this.currentState);
     this.executeActions(stateNodes.reduce((acc, node) => acc.concat(node.onEntry), []), null, 'tutorial');
     if (this.storageHasKey(STORAGE_KEYS.FEATURE_LIST)) {
       this.set('featureList', this.getExtState(STORAGE_KEYS.FEATURE_LIST));
       if (this.storageHasKey(STORAGE_KEYS.FEATURE_STATE)) {
+        if (this.storageHasKey(STORAGE_KEYS.FEATURE_STATE_HISTORY)) {
+          this.set('featureMachineHistory', this.getExtState(STORAGE_KEYS.FEATURE_STATE_HISTORY));
+        }
```

Comment:
We should look at breaking this up / simplifying it soon ... if's 3 deep is getting more difficult to read.